lib/assessment.js

```diff
@@ -8,25 +8,80 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
 
+var min = function min(previousValue, currentValue, index, array) {
+  return previousValue < currentValue ? previousValue : currentValue;
 };
 
+console.log(reduce(numbers, min));
+console.log(numbers.reduce(function (previousValue, currentValue) {
+  return previousValue < currentValue ? previousValue : currentValue;
+}));
+
+
+};
+
+// I used Antony's Max Example of code and switched the operators.
+// I do not think I could logic out the whole thing as concisely as this.
+// I cannot write this code from scratch myself but I can read this and edit it
+
+
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
+
 var sum = function sum() {
 
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
+};
+
+var numbers = [13, 26, 32, 5];
+
+console.log(reduce(numbers, sum));
+console.log(numbers.reduce(function (previousValue, currentValue) {
+  return previousValue + currentValue;
+}));
+
+
 };
 
+// Same case for solving here as above-
+// I heavily relied on Antony's code and can readi it pretty well
+// I Feel rushed so maybe I am not catching all of the edits but
+// I am concerned that I cannot construct this code myself,
+//I hoep i chose the correct code to model from, even.
+// I do understand how javascript is fucntioning though, most of the time.
+
+
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
 
+  var any = function any(array, callback, thisArg) {
+  // callback(currentValue, index, array)
+
+  //check for true first
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+    // if callback ever returns false, result will be set
+    // to false amd the callback will not be called in any
+    // subsequent iterations.
+    result = result && callback(currentValue, index, array);
+  });
+
+  return result;
 };
```

Comment:
Here is an explanation for how to turn `every` into `any`:

The question asks for a function that returns `true` if the callback returns `true` for **any** element in the array. This suggests that we should be using a logical OR (`||`), not a logical AND (`&&`). A logical AND would be suitable for determining whether a callback returns `true` for **all** elements of the array. 

To know what our initial value of `result` should be, we consult the [truth table for the OR operation](https://en.wikipedia.org/wiki/Logical_disjunction#Truth_table). As we see on row 2 of the truth table, `true || false` evaluates to `true`. Therefore, if we initialize `result` to `true`, `any` will return `true` regardless of what the callback returns for each element of the list -- this is not what we want. There's only one more value to consider: `false`.

We see on lines 4 and 3 of the truth table that `false || false` yields `false`, and `false || true` yields `true`. So, if `result` starts out as `false`, it will remain `false` until it is ORed with `true`. This sounds like what we want. Also, once `result` stores `true`, it will remain `true` no matter what we OR it with. This also sounds like what we want here.

So line 42 should read: `var result = false;` and line 45 should read: `result = result || callback(...);`

3/4 credit.